src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int COEFFICIENT_GROW = 2;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] map;
+    private int size;
+
+    public MyHashMap() {
+        map = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 >= map.length * LOAD_FACTOR) {
+            resizeMap();
+        }
+        putItem(map, key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getBucketIndex(key, map.length);
```

Comment:
```suggestion
        int index = getBucketIndex(key);
```
isn't map.length available in any method?

Reply:
ok